File: CSharp/ByTier/C/1873_C_Target_Practice.cs
// Problem: CF 1873 C - Target Practice
// https://codeforces.com/contest/1873/problem/C

/*
 * Purpose: Calculate the total points scored on a 10x10 target based on the placement of arrows.
 *          The target has 5 concentric rings with point values from 1 to 5.
 *          Arrows are counted based on which ring they land in.
 *
 * Algorithms/Techniques:
 *   - Direct ring detection by checking positions in predefined rings
 *   - Iterative scanning of grid edges to identify arrows and assign point values accordingly
 *
 * Time Complexity: O(t * n^2) where t is the number of test cases and n=10 (fixed grid size)
 * Space Complexity: O(n^2) for storing the grid of size 10x10 per test case
 */

using System;
class Program {
  static void Main(string[] args) {
    double E = 1e-6;
    double PI = 3.1415926535;
    double INF = 1e18;
    double MOD = 1e9 + 7;
    int t = int.Parse(Console.ReadLine());
    while (t > 0) {
      int n = 10;
      string[] vec = new string[n];
      for (int i = 0; i < n; i++) {
        vec[i] = Console.ReadLine();
      }
      int sum = 0;
      // Check outermost ring (1 point) - top and bottom edges
      for (int i = 0; i < 10; i++) {
        if (vec[0][i] == 'X') // Top edge
          sum += 1;
        if (vec[9][i] == 'X') // Bottom edge
          sum += 1;
      }
      // Check second ring (1 point) - left and right edges
      for (int i = 1; i < 9; i++) {
        if (vec[i][0] == 'X')
          sum += 1;
        if (vec[i][9] == 'X')
          sum += 1;
      }
      // Check third ring (2 points)
      for (int i = 1; i < 9; i++) {
        if (vec[1][i] == 'X')
          sum += 2;
        if (vec[8][i] == 'X')
          sum += 2;
      }
      // Check fourth ring (2 points)
      for (int i = 2; i < 8; i++) {
        if (vec[i][1] == 'X')
          sum += 2;
        if (vec[i][8] == 'X')
          sum += 2;
      }
      // Check fifth ring (3 points)
      for (int i = 2; i < 8; i++) {
        if (vec[2][i] == 'X')
          sum += 3;
        if (vec[7][i] == 'X')
          sum += 3;
      }
      // Check sixth ring (3 points)
      for (int i = 3; i < 7; i++) {
        if (vec[i][2] == 'X')
          sum += 3;
        if (vec[i][7] == 'X')
          sum += 3;
      }
      // Check seventh ring (4 points)
      for (int i = 3; i < 7; i++) {
        if (vec[3][i] == 'X')
          sum += 4;
        if (vec[6][i] == 'X')
          sum += 4;
      }
      // Check eighth ring (4 points)
      for (int i = 4; i < 6; i++) {
        if (vec[i][3] == 'X')
          sum += 4;
        if (vec[i][6] == 'X')
          sum += 4;
      }
      // Check center ring (5 points)
      for (int i = 4; i < 6; i++) {
        if (vec[4][i] == 'X')
          sum += 5;
        if (vec[5][i] == 'X')
          sum += 5;
      }
      Console.WriteLine(sum);
      t--;
    }
  }
}


// https://github.com/VaHiX/CodeForces/

File: CPP/ByTier/C/1873_C_Target_Practice.cpp
// Problem: CF 1873 C - Target Practice
// https://codeforces.com/contest/1873/problem/C

/*
C. Target Practice
time limit per test1 second
memory limit per test256 megabytes

A 10×10 target is made out of five "rings" as shown. Each ring has a different point value: 
the outermost ring — 1 point, the next ring — 2 points, ..., the center ring — 5 points.
Vlad fired several arrows at the target. Help him determine how many points he got.

Algorithms/Techniques:
- For each arrow ('X') on the grid, compute its Manhattan distance to the nearest edge of the 10x10 grid.
- Based on that distance, calculate which ring it belongs to (ring 1 to 5).
- Sum up the point values accordingly.

Time Complexity: O(t * B^2) where B=10 (constant), so effectively O(t)
Space Complexity: O(B^2) for storing the 10x10 grid per test case.

Input:
The input consists of multiple test cases. The first line of the input contains a single integer
t (1≤t≤1000) — the number of test cases.
Each test case consists of 10 lines, each containing 10 characters. Each character in the grid is either
X (representing an arrow) or . (representing no arrow).

Output:
For each test case, output a single integer — the total number of points of the arrows.

Example:
input
4
X.........
..........
.......X..
.....X....
......X...
..........
.........X
..X.......
..........
.........X
..........
..........
..........
..........
..........
..........
..........
..........
..........
..........
..........
..........
..........
..........
....X.....
..........
..........
..........
..........
..........
XXXXXXXXXX
XXXXXXXXXX
XXXXXXXXXX
XXXXXXXXXX
XXXXXXXXXX
XXXXXXXXXX
XXXXXXXXXX
XXXXXXXXXX
XXXXXXXXXX
XXXXXXXXXX
output
17
0
5
220
*/

#include <stdio.h>
#include <iostream>
#include <vector>
#include <string>

int main() {
  const int B = 10;
  std::ios_base::sync_with_stdio(false);
  long t;
  std::cin >> t;
  while (t--) {
    std::vector<std::string> m(B); // Store the 10x10 grid
    for (long p = 0; p < B; p++) {
      std::cin >> m[p]; // Read each row of the grid
    }
    long total(0);
    for (long row = 0; row < B; row++) {
      for (long col = 0; col < B; col++) {
        if (m[row][col] == '.') {
          continue; // Skip empty cells
        }
        // Find the minimum distance to any edge (row or column)
        long xrow = (row < B - 1 - row) ? row : (B - 1 - row);
        long xcol = (col < B - 1 - col) ? col : (B - 1 - col);
        // The score is 1 plus the minimum of the two distances
        long score = 1 + (xrow < xcol ? xrow : xcol);
        total += score;
      }
    }
    printf("%ld\n", total);
  }
}


// https://github.com/VaHiX/codeForces/